wgengine/router_linux.go

```diff
@@ -160,13 +166,24 @@ func (r *linuxRouter) SetRoutes(rs RouteSettings) error {
 
 func (r *linuxRouter) Close() error {
 	var ret error
-	if err := r.restoreResolvConf(); err != nil {
-		r.logf("failed to restore system resolv.conf: %v", err)
-		if ret == nil {
+	set := func(err error) {
+		if ret == nil && err != nil {
 			ret = err
 		}
 	}
-	// TODO(apenwarr): clean up iptables etc.
+	if err := r.restoreResolvConf(); err != nil {
+		r.logf("failed to restore system resolv.conf: %v", err)
+		set(err)
+	}
+	set(r.ipt4.Delete("filter", "FORWARD",
```

Comment:
This will wipe forwarding rules installed by !us, if they happen to match. Suggest adding `-m comment --comment "tailscale"` to all inserted rules, so that we always match exactly on things we create, and not random other firewall config.

Reply:
(bonus, it documents to users where these rules came from in `iptables -L` output)

Reply:
Neat! Done.